core_contracts/fees/fees.py

```diff
@@ -155,16 +155,22 @@ def tokenFallback(self, _from: Address, _value: int, _data: bytes) -> None:
                 path = json_loads(self._routes[self.msg.sender][self._getContractAddress("baln")])
             except:
                 path = []
-                
-            if path:
-                # Use router.
-                self._transferToken(self.msg.sender, self._getContractAddress("router"), 
-                                    self._getTokenBalance(self.msg.sender), 
-                                    self._createDataFieldRouter(self._getContractAddress("dividends"), path))
-            else:
-                # Use dex.
-                self._transferToken(self.msg.sender, self._getContractAddress("dex"), self._getTokenBalance(self.msg.sender), 
-                                    self._createDataFieldDex(self._getContractAddress("baln"), self._getContractAddress("dividends")))
+
+            # Attempt to process fees. Catch Error and report via eventlog if failure.
+            try:
+                if path:
+                    # Use router.
+                    self._transferToken(self.msg.sender, self._getContractAddress("router"), 
+                                        self._getTokenBalance(self.msg.sender), 
+                                        self._createDataFieldRouter(self._getContractAddress("dividends"), path))
+
+                else:
+                    # Use dex.
+                    self._transferToken(self.msg.sender, self._getContractAddress("dex"), self._getTokenBalance(self.msg.sender), 
+                                        self._createDataFieldDex(self._getContractAddress("baln"), self._getContractAddress("dividends")))
+            
+            except BaseException as e:
+                self.FeeNotProcessed(self.msg.sender, repr(e))
```

Comment:
Should we return after event is sent? If not we will reset the timer for next fee processing, resulting in not getting this error again until the its time for next fee processing.

Reply:
I think this looks good. Even if we do not catch it right away it will not be so bad if some fees accumulate on the FeeHandler. I think we can also have the backend watch for this event and alert us. For now this will keep the platform from getting stuck if there is an unforeseen problem.

Reply:
Sounds good Dan. Thanks for reviewing.